homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -71,10 +81,65 @@ def get_platforms(config_entry):
         for air_monitor_model in MODELS_AIR_MONITOR:
             if model.startswith(air_monitor_model):
                 return AIR_MONITOR_PLATFORMS
-
+    _LOGGER.error(
+        "Unsupported device found! Please create an issue at "
+        "https://github.com/syssi/xiaomi_airpurifier/issues "
+        "and provide the following data: %s",
+        model,
+    )
     return []
 
 
+async def async_create_miio_device_and_coordinator(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+):
+    """Set up a data coordinator and one miio device to service multiple entities."""
+    model = entry.data[CONF_MODEL]
+    host = entry.data[CONF_HOST]
+    token = entry.data[CONF_TOKEN]
+    name = entry.title
+    device = None
+    if model not in MODELS_HUMIDIFIER:
+        return
+    if model in MODELS_HUMIDIFIER_MIOT:
+        device = AirHumidifierMiot(host, token)
+    else:
+        device = AirHumidifier(host, token, model=model)
+
+    # removing fan platform entity for humidifiers
+    entity_registry = await en.async_get_registry(hass)
+    entity_id = entity_registry.async_get_entity_id("fan", DOMAIN, entry.unique_id)
+    if entity_id:
+        entity_registry.async_remove(entity_id)
+
+    async def async_update_data():
+        """Fetch data from the device using async_add_executor_job."""
+        # On state change the device doesn't provide the new state immediately.
+        try:
+            async with async_timeout.timeout(10):
+                return await hass.async_add_executor_job(device.status)
+
+        except DeviceException as ex:
+            _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+    # Create update miio device and coordinator
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name=name,
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=60),
+    )
+    hass.data[DOMAIN][entry.entry_id] = {
+        KEY_DEVICE: device,
+        KEY_COORDINATOR: coordinator,
+    }
+    # Trigger first data fetch
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
This will fail the whole integration setup and retry it later. Is that what we want even if it's just one device that isn't connected ok?

Reply:
But the coordinator is created for config entry, that is for one device. Am I missing something?

Reply:
The setup is repeated for every device in the network. It should fail if no data could be fetched.